Task_2_1

Sharapov-Maksim opened the pull request:
First version of task

Sharapov-Maksim committed:
First commit

First version of task

vlitomsk commented on Task_2_1/src/main/java/Stack.java:
```diff
@@ -0,0 +1,79 @@
+
+            @Override
+            public T next() /*throws IndexOutOfBoundsException*/ {
+                //if (!hasNext()) throw new IndexOutOfBoundsException("End of list.");\
```
Исключение хорошо бы выбросить, но в интерфейсе немного другое прописано: https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#next--

vlitomsk commented on Task_2_1/src/main/java/Stack.java:
```diff
@@ -0,0 +1,79 @@
+     * @throws NegativeArraySizeException when user try to pop empty stack
+     */
+
+    public T pop () throws Exception{
```
Лучше в throws писать максимально конкретный класс.
Почему так лучше?

Sharapov-Maksim replied:
Видимо так проще ловить и обрабатывать бросаемые исключения. Ведь в случае с throws Exception мы лишь даём  информацию, что что-то может пойти не так, но не конкретные исключения.

vlitomsk commented on Task_2_1/src/test/java/StackTest.java:
```diff
@@ -0,0 +1,89 @@
+import static org.junit.Assert.*;
+
+public class StackTest {
+
```
Предлагаю добавить тест, проверяющий случай выталкивания элемента из пустого стека.
Для проверки того, что тест выбросил исключение, в junit есть специальная аннотация.

vlitomsk replied:
Примеры тут есть: https://www.guru99.com/junit-annotations-api.html

vlitomsk reviewed (changes requested):
Замечания нужно исправить, на вопросы ответить %)

Sharapov-Maksim committed:
Fixes

Изменены бросаемые исключения и добавлены тесты для проверки бросаемых исключений.

vlitomsk reviewed (changes requested):
ОК принято
